### ISSUE TITLE
FR: AssetManager support for choosing file/dir to be publish

### ISSUE THREAD

**Issue by mdmunir** [yiisoft/yii2#8566]
Suggest add option `except` and `only` on `AssetManager`.

```php
protected function publishDirectory($src, $options)
{
    $dir = $this->hash($src . filemtime($src));
    $dstDir = $this->basePath . DIRECTORY_SEPARATOR . $dir;
    if ($this->linkAssets) {
        if(empty($options['only']) && empty($options['except']){
            if (!is_dir($dstDir)) {
                symlink($src, $dstDir);
            }
        }else{
            $options['recursive'] = false;
            foreach(FileHelper::findFiles($src,$options) as $file){
                $base = basename($file);
                if(!is_dir($dstDir.'/'.$base) && !is_dir($dstDir.'/'.$base)){
                    symlink($file, $dstDir.'/'.$base);
                }
            }
        }
    } elseif (!empty($options['forceCopy']) || ($this->forceCopy && !isset($options['forceCopy'])) || !is_dir($dstDir)) {
        $opts = [
            'dirMode' => $this->dirMode,
            'fileMode' => $this->fileMode,
        ];
        if(!empty($options['only'])){
            $opts['only'] = $options['only'];
        }
        if(!empty($options['except'])){
            $opts['except'] = $options['except'];
        }


```

**Comment by klimov-paul** [comment]
It is rather strange to have redundant files under asset source directory.
Which files or directories publishing you try to avoid using such option? Can you provide rael-life example?

**Comment by mdmunir** [comment]
look at [adminlte](https://github.com/almasaeed2010/AdminLTE/tree/1.3.0)(old version). They has no `dist` folder to publish. But we dont want to publish entire directory. We dont want to publish pages, ajax, less, etc. We also cannot publish each folder as individual asset because they has relative link in css (`url('../img/ajax-loader1.gif')`). If you understand what i mean :D

**Comment by klimov-paul** [comment]
Relates #7307

**Comment by klimov-paul** [comment]
Issue resolved by commit 8a506e1
